test/lang_utils.h

```diff
@@ -0,0 +1,110 @@
+/**
```

Comment:
This extraction is useful by itself and independent, please split in a separate commit